.github/workflows/dart_native.yaml

```diff
@@ -0,0 +1,64 @@
+# Native Test runs code on non-Linux platforms
+name: Dart (Native)
+on:
+  workflow_call:
+    inputs:
+      working-directory:
+        description: The working directory relative to the repo root
+        required: true
+        type: string
+
+jobs:
+  native_test:
+    name: Native Test
+    timeout-minutes: 60
+    strategy:
+      # Allows other matrix items to run if one fails
+      fail-fast: false
+      matrix:
+        os:
+          # macos-latest currently points to macos-11
+          # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on
+          - macos-12
```

Comment:
callout (no change needed inthis PR): This is going to consume lots of macos compute across its consumers, so just might be something to keep an eye on if we hit a limit at some point. Also dk if people run workflows on their fork like I do where you can only run 5 mac jobs at once.

Reply:
Good point. My thinking with these is that they will only run on pushes (not pull requests) so they can run asynchronously without blocking any real work. Open to further discussion, though (maybe run them on a schedule instead?)